### PR TITLE
Feature/clean up reflection warnings

### PR DESCRIPTION
**Pull request by samiwh**
Removed injection code where it wasn't needed and suppressed warnings were it was.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs [hunk 1]
```diff
@@ -70,37 +46,25 @@ protected override void OnDestroyManager()
...
 
-        // todo replace this with something that polls for entities added and removed
         protected override unsafe void OnUpdate()
         {
-            for (var i = 0; i < addedEntitiesData.Length; i++)
+            foreach (var entityId in entitySystem.GetEntitiesAdded())
             {
-                var entity = addedEntitiesData.Entities[i];
-                var spatialEntityId = addedEntitiesData.SpatialEntityIds[i].EntityId;
+                workerSystem.TryGetEntity(entityId, out var entity);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs]: Is this intended to discard the result of the `TryGet`? 

If so is it worth exposing a `GetEntity` method which doesn't have the out param and will actually throw if the entity doesn't exist?

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs]: Such that the use cases of the methods break down into: 

- `GetEntity`: Entity will be there and its an error if its not.
- `TryGetEntity`: Entity may be there and I can handle the case where its not.

File: workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs [hunk 2]
```diff
@@ -70,37 +46,25 @@ protected override void OnDestroyManager()
...
-                PostUpdateCommands.AddComponent(entity, new InitializedEntitySystemState
-                {
-                    EntityId = spatialEntityId
-                });
             }
 
-            EntityId* entitiesToRemove = stackalloc EntityId[removedEntitiesData.Length];
-            for (var i = 0; i < removedEntitiesData.Length; i++)
+            var removedEntities = entitySystem.GetEntitiesRemoved();
+            foreach (var entityId in removedEntities)
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs]: I enjoy how much more concise this is now 💯

File: workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs]: Yes and maybe. The entity definitely exists at this point, however mid term I'd rather look at removing the dependency on ecs entities. GetEntity should be a method though.

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs]: Basically I didn't touch that stuff too much as I suspect we may want to change it a lot.

**Pull request merged**